### ISSUE TITLE
Different behavior in optimized padding block

### ISSUE THREAD

**Issue by MikeAleksa** [nebuly-ai/nebullvm#67]
I'm trying to optimize a CNN model and getting different behavior in a padding/concatenation block during optimization, causing the optimization to fail. The block in question:

```
class PadConcatBlock(nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self, down_layer, x):
        x1_shape = down_layer.shape
        x2_shape = x.shape

        height_diff = (x1_shape[2] - x2_shape[2]) // 2

        # if down_layer is larger than x, pad down_layer
        down_layer_padded = F.pad(
            down_layer,
            (0, 0, abs(height_diff), 0, 0, 0),
            mode="replicate",
        )

        x = torch.concat([down_layer_padded, x], axis=1)
        return x
```

down_layer shape is `torch.Size([1, 32, 1024, 16])`
x shape is `torch.Size([1, 32, 1025, 16])`

height_diff should be `-1` but when I log height_diff during the `Running Optimization using ONNX interface` step, it sometimes comes back as `-1` and sometimes comes back as `tensor(0)`, causing torch.concat() to fail.

Why would the behavior of `height_diff = (x1_shape[2] - x2_shape[2]) // 2` change?

**Comment by valeriosofi** [comment]
Hi @MikeAleksa, probably the error is due to the fact that python's floor operator (//) is not well supported by onnx, to investigate your problem further we would ask you to please send us the code where you make the call to the nebullvm API to do the optimization and also the complete model code in case the block you send us before is not complete. Btw there I notice that you assume that down_layer is always smaller than x, be careful because if it were larger it would still perform padding leading to an incorrect shape because of abs().
